[Toh Bing Cheng] IP

bingcheng45 opened the pull request:
DukePro

> “Sonic and Tails are Best of Friends.” – DUKE NUKEM ([source](https://twitter.com/quotesofduke?lang=en))

DukePro frees your mind of having to remember things you need to do. It's,

text-based
easy to learn
~~FAST~~ SUPER FAST to use
All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2122S1/ip/pull/368).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

bingcheng45 committed:
Greet, Echo, Exit

bingcheng45 committed:
Add, List

bingcheng45 committed:
Added ability to mark items as done, new state in Task.

bingcheng45 committed:
Added feature TODO, Deadline & Event, with respective task events

bingcheng45 committed:
Added automated Text UI Testing with expected values.

bingcheng45 committed:
Added exception handling and more user friendly error messages.

bingcheng45 committed:
Added ability to delete task from the list.

bingcheng45 committed:
Added Enums for available commands for the program to help with the scope.

bingcheng45 committed:
Added ability to read and write files.

bingcheng45 committed:
Improve code readability

bingcheng45 committed:
Add localDate functionality

bingcheng45 committed:
Merge branch 'branch-Level-7'

bingcheng45 committed:
Resolve merge conflict

bingcheng45 committed:
Refactor code to be more OOP

bingcheng45 committed:
Move respective class files into their packages

bingcheng45 committed:
Add JUnit test for command and task class

bingcheng45 committed:
Build jar package

bingcheng45 committed:
Add JavaDoc to the core programs

bingcheng45 committed:
Conforms to coding standards

bingcheng45 committed:
Add Find functionality

bingcheng45 committed:
Merge branch 'branch-A-JavaDoc'

bingcheng45 committed:
Merge branch 'branch-A-CodingStandard'

bingcheng45 committed:
Resolve merge conflict

bingcheng45 committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

bingcheng45 committed:
Add gradle support

JasonC01 commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,34 @@
+package duke.command;
+
+import duke.*;
```
I believe wildcard import should not be used at all times

JasonC01 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,65 @@
+ */
+public abstract class Task {
+    private String value = null;
+    private boolean done = false;
```
isDone seems to be a better name for booleans

JasonC01 reviewed (commented):
Looks really good
It was hard to look for any flaws in the code. Great job!!:)

bingcheng45 committed:
Accomodate with the checkstyle guidelines.

bingcheng45 committed:
Merge branch 'branch-A-Gradle'

pattan9955 commented on build.gradle:
```diff
@@ -0,0 +1,47 @@
+    toolVersion = '8.29'
+}
+
+run{
```
Should there be a space between the run and the curly brace?

pattan9955 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,66 @@
+     */
+    public void runProgram() {
+        ui.showWelcome();
+        boolean run = true;
```
Should this be named isRunning in accordance with the coding guidelines?

pattan9955 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,124 @@
+     * @throws IncompleteCommandException if command is incomplete
+     */
+    public static Command parse(String input) throws IncompleteCommandException {
+        if (input.toUpperCase().equals(CommandList.BYE.toString())) {
```
Should the input.toUpperCase() in lines 28, 30, 32, 36, 55 and 73 be put into a separate line to allow for better readability?

pattan9955 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,179 @@
+        for (int i = 0; i < taskList.getTotalNumberOfTask(); i++) {
+            Task currTask = taskList.getTaskById(i);
+            JSONObject taskDetails = new JSONObject();
+            String type = currTask.getType().toString();
```
Should there be a newline between lines 90 and 91 to separate the code into separate logical chunks?

pattan9955 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,179 @@
+        JSONObject taskObject = (JSONObject) task.get("task");
+
+        String value = (String) taskObject.get("value");
+        boolean done = (boolean) taskObject.get("done");
```
Should this be renamed isDone in accordance with the coding standard?

pattan9955 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,179 @@
+        boolean done = (boolean) taskObject.get("done");
+        String time = (String) taskObject.get("time");
+        String type = (String) taskObject.get("type");
+        type = type.toUpperCase();
```
I like that you abstracted out the conversion of the type to UpperCase which allowed for cleaner looking code in lines 165-171! :D

pattan9955 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,100 @@
+    /**
+     * Display goodbye message.
+     */
+    public void goodBye() {
```
Should this method be renamed showGoodBye() to be consistent with the showWelcome() method?

bingcheng45 replied:
yeap good catch! I have made changes to it already. Thank you.

pattan9955 reviewed (commented):
Overall, LGTM! The code is generally quite clean and rather readable, with only a few lines that were possibly a bit too long. Apart from some minor naming and coding standard violations, well done! :)

bingcheng45 committed:
Add GUI

bingcheng45 committed:
Implemented GUI with Fxml and Varargs

bingcheng45 committed:
Merge branch 'branch-Level-10'

bingcheng45 commented:
DukePro

> “Sonic and Tails are Best of Friends.” – DUKE NUKEM ([source](https://twitter.com/quotesofduke?lang=en))

DukePro frees your mind of having to remember things you need to do. It's,

text-based
easy to learn
~~FAST~~ SUPER FAST to use
All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2122S1/ip/pull/368).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

bingcheng45 committed:
Port over duke functionality onto the UI

This was needed as the methods previously printed the messages but we
needed to pass it over

The current implementation would allow for messages to be passed
around.Port over duke functionality onto the UI

This was needed as the methods previously printed the messages but we
needed to pass it over

The current implementation would allow for messages to be passed around.

bingcheng45 committed:
Add assertions to ensure assumptions are in check.

Most assertion are for checking null.

bingcheng45 committed:
Improve code quality in Parser

bingcheng45 committed:
Modify to use streams.

bingcheng45 committed:
Add assertions

bingcheng45 committed:
Improve code quality

bingcheng45 committed:
Merge pull request #1 from bingcheng45/branch-A-CodeQuality

Improve code quality

bingcheng45 committed:
Merge pull request #2 from bingcheng45/branch-A-Assertions

Add assertions

bingcheng45 committed:
Improve search to allow partial words for find command

bingcheng45 committed:
Merge pull request #3 from bingcheng45/branch-C-BetterSearch

Improve search to allow partial words for find command

bingcheng45 committed:
Set up CI

bingcheng45 committed:
Merge branch 'master' of https://github.com/bingcheng45/ip

bingcheng45 committed:
Improve Gui

bingcheng45 committed:
Add title name

bingcheng45 committed:
Modified Readme.md

bingcheng45 committed:
Set theme jekyll-theme-slate

bingcheng45 committed:
Set theme jekyll-theme-slate

bingcheng45 committed:
Fix spelling mistakes

bingcheng45 committed:
Merge branch 'master' of https://github.com/bingcheng45/ip

bingcheng45 committed:
SLAP methods

bingcheng45 committed:
Fix naming and standardise code
